components/merkledb/src/validation.rs

```diff
@@ -12,26 +23,43 @@ pub fn is_valid_index_name<S: AsRef<str>>(name: S) -> bool {
 /// Check that latin1 character is allowed in index name.
 /// Only these combination of symbols are allowed:
 ///
-/// `[0..9]`, `[a-z]`, `[A-Z]`, `_`, `-`, `.`
+/// `[0..9]`, `[a-z]`, `[A-Z]`, `_`, `-`
 pub fn is_allowed_latin1_char(c: u8) -> bool {
     match c {
-        48..=57   // 0..9
+        48..=57     // 0..9
         | 65..=90   // A..Z
         | 97..=122  // a..z
-        | 45..=46   // -.
+        | 45        // -
         | 95        // _
         => true,
         _ => false,
     }
 }
 
 /// Calls the `is_valid_name` function with the given name and panics if it returns `false`.
-pub(crate) fn assert_valid_name<S: AsRef<str>>(name: S) {
+pub(crate) fn assert_index_valid_name<S: AsRef<str> + fmt::Debug>(name: S) {
```

Comment:
Nit: a private helper HOF taking a predicate can be introduced here to avoid code duplication.